### ISSUE TITLE
[Bug]: Build failure on Mac OS 14.0

### ISSUE THREAD

**Issue by ahmedriza** [abseil/abseil-cpp#1558]
### Describe the issue

On Mac OS 14.0, there are two test failures. 
```
//absl/meta:type_traits_test                                    FAILED TO BUILD
//absl/strings:str_format_convert_test                          FAILED TO BUILD
```

The specific failures are:
```
absl/meta/type_traits_test.cc:770:3: error: static assertion failed due to requirement '!absl::is_trivially_relocatable<S>::value':
  static_assert(!absl::is_trivially_relocatable<S>::value, "");
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
absl/strings/internal/str_format/convert_test.cc:1251:45: error: code will never be executed [-Werror,-Wunreachable-code]
  const NativePrintfTraits &native_traits = VerifyNativeImplementation();
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Steps to reproduce the problem

Invoked the build on Mac OS:
```
./ci/macos_xcode_bazel.sh
```

### What version of Abseil are you using?

```
git rev-parse HEAD
89d2caa104563ad8c2182282f5d4676f6747bd0e
```
as well as `LTS 20230802.1`

### What operating system and version are you using?

Mac OS 14.0

### What compiler and version are you using?

```
clang --version

Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin23.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

### What build system are you using?

```
bazel --version
bazel 6.4.0-homebrew
```

### Additional context

```
xcode-select --version
xcode-select version 2399.
```

**Comment by Quuxplusone** [comment]
My understanding is that this would be fixed (on the Clang side) by https://github.com/llvm/llvm-project/pull/84621 ; please take a look at that PR and leave a comment or emoji or something if you'd like Clang to do it!

On the Abseil side, I believe this was fixed by opting out of Clang's `__is_trivially_relocatable(T)` on Apple platforms in df2c771ec596b385448117f237ee70be35efe4ce, and it's _certainly_ fixed as of 14b8a4eac3e5a7b97ba4cc7b7dadf2a85aae8215 which abandons the buggy builtin entirely (for now). So there's nothing more to do on the Abseil side AFAIK. Abseil would _like_ to re-enable the use of Clang's `__is_trivially_relocatable(T)`, but that's impossible until Clang merges https://github.com/llvm/llvm-project/pull/84621 .

**Comment by derekmauro** [comment]
Yes, this can be closed. Thanks.
